Automattic/sensei
# Add home tasks backend basic structure.

Fixes #5739 

### Changes proposed in this Pull Request
* [x] Add Tasks structure in the backend.
* [x] Implement sample "Site Setup" task.
* [x] Integrate provider with mapper and REST API.
* [ ] ~Add logic based on purposes to decide on tasks.~ Will be implemented on each separate task.
* [x] Allow for other plugins to modify returned tasks.
* [ ] ~All the individual tasks~ Will be implemented in separate tickets.

### Testing instructions
- Similar to previous tasks (#5739 #5741 #5627 ).
- Check that the "Setup Site" task is returned.

### Screenshot / Video
```
$ curl -s -X GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .tasks
{
  "items": {
    "setup-site": {
      "title": "Set up Course Site",
      "priority": 100,
      "url": "https://test2.local/wp-admin/admin.php?page=sensei_setup_wizard",
      "image": null,
      "done": true
    }
  }
}
```
